src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -184,6 +205,28 @@ public void Reanalyze(IIncrementalAnalyzer analyzer, ReanalyzeScope scope, bool
                 }
             }
 
+            private void OnActiveDocumentChanged(object sender, DocumentId activeDocumentId)
+            {
+                IAsyncToken asyncToken;
+                if (SolutionCrawlerOptions.GetBackgroundAnalysisScope(_registration.Workspace.Options) == BackgroundAnalysisScope.ActiveFile &&
+                    activeDocumentId != null)
+                {
+                    var activeDocument = _registration.Workspace.CurrentSolution.GetDocument(activeDocumentId);
```

Comment:
workspace options and workspace.currentsolution are both mutable (and can change separately).  if you're on the bg, can you read these out of sync?  why not just get CurrentSolution, and access options off of it?  that way you're certain the solution snapshot and options snapshot correspond to each other.